snprc_ehr/src/org/labkey/snprc_ehr/history/LabworkDataSource.java

```diff
@@ -82,6 +82,7 @@ protected String getHtml(Container c, Results rs, boolean redacted) throws SQLEx
         if (rs.hasColumn(FieldKey.fromString("tissue")) && rs.getObject("tissue") != null)
             sb.append("Sample/Tissue: ").append(snomedToString(rs, FieldKey.fromString("tissue"), FieldKey.fromString("tissue/meaning")));
 
+        sb.append(safeAppend(rs, "Speicmen Id", "sampleId"));
```

Comment:
I did checked for the safeAppend method but could not find it. Can i know from where this safeAppend method is implemented?

Reply:
org.labkey.api.ehr.history.AbstractDataSource#safeAppend